src/common/dto/transaction-options.dto.ts

```diff
@@ -0,0 +1,35 @@
+/* istanbul ignore file */
+
+import { ApiHideProperty, ApiProperty } from '@nestjs/swagger';
+import { IsEnum, IsObject, IsOptional, IsString, IsUrl, ValidateIf } from 'class-validator';
+
+import { ProcessMode } from '~/common/types';
+
+export class TransactionOptionsDto {
+  @ApiProperty({
+    description: 'An identifier or address for the account that should sign the transaction',
+    example: 'alice',
+  })
+  @IsString()
+  readonly signer: string;
+
+  @ApiProperty({
+    description: 'How the transaction should be processed',
+    enum: ProcessMode,
+  })
+  @IsEnum(ProcessMode)
+  readonly processMode: ProcessMode;
+
+  // Hide the property so the interactive examples work without additional setup
+  @ApiHideProperty()
+  @ValidateIf(({ processMode }) => processMode === ProcessMode.SubmitWithCallback)
+  @IsString()
+  @IsUrl()
```

Comment:
`@IsOptional`?

Reply:
If someone uses callback, then they have to provide a webhook URL. I'll add a note